File: Assets/Code/Outfit7/Raven/TrackComponents/AnimationData/CurveSpecialization/RavenAnimationDataCurveDouble.cc
#ifdef STARLITE
#include "RavenAnimationDataCurveDouble.h"
#include "RavenAnimationDataCurveDouble.cs"

namespace Starlite {
    namespace Raven {
        RavenAnimationDataCurveDouble::RavenAnimationDataCurveDouble() {
        }

        Double RavenAnimationDataCurveDouble::EvaluateAtTime(double time, double duration) {
            return (Double)m_Curves[0]->Evaluate((float)GetNormalizedTime(GetCurrentTime(time, duration), duration, m_EaseType));
        }

        void RavenAnimationDataCurveDouble::SetStartingValues(const Double& values) {
            m_Curves = Array<Ref<AnimationCurve>>(1);
            m_Curves[0] = Object::New<AnimationCurve>();
            m_Curves[0]->AddKeyframe(AnimationCurve::Keyframe(0, (float)values));
            m_Curves[0]->AddKeyframe(AnimationCurve::Keyframe(1, (float)values));
        }

        void RavenAnimationDataCurveDouble::SyncStartingValues(const Double& values) {
            auto& curve = m_Curves[0];
            auto& key = curve->GetKeyframes()[0];
            key.value = (float)values;
        }
    } // namespace Raven
} // namespace Starlite
#endif

File: Assets/Code/Outfit7/Raven/TrackComponents/AnimationData/CurveSpecialization/RavenAnimationDataCurveDouble.cs
//{ "engine_register_objects":["Starlite__Raven__RavenAnimationDataCurveDouble"] }
#if !__cplusplus
#pragma warning disable
namespace Starlite {

    namespace Raven {
    
        // RavenAnimationDataCurveDouble
        [Starlite.ObjectClassAttribute("Starlite::Raven::RavenAnimationDataCurveDouble")]
        public sealed partial class RavenAnimationDataCurveDouble : global::Starlite.Raven.RavenAnimationDataCurve<double> {
        }
        
    }
    
}
#pragma warning restore
#else
// RavenAnimationDataCurveDouble
namespace Starlite { namespace Raven { 
SIMPLEMENT_CLASS_WITH_PROPERTIES_IN_CLASS_SCOPED(Starlite::Raven::RavenAnimationDataCurveDouble, RavenAnimationDataCurveDouble, RavenAnimationDataCurveDouble, Starlite::Raven::RavenAnimationDataComponentBase, Starlite::SceneObjectComponent, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataCurveDouble, m_Curves, m_Curves, kArray, kObject, Starlite::AnimationCurve::TypeId, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataCurveDouble, m_EaseType, m_EaseType, kInt, kUnknown, -1, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataCurveDouble, m_ValueType, m_ValueType, kInt, kUnknown, -1, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataCurveDouble, m_RepeatCount, m_RepeatCount, kInt, kUnknown, -1, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataCurveDouble, m_Mirror, m_Mirror, kBool, kUnknown, -1, "")
SIMPLEMENT_CLASS_END(RavenAnimationDataCurveDouble)
} } 
SIMPLEMENT_CLASS_SCOPE(Starlite__Raven__RavenAnimationDataCurveDouble, Starlite::Raven, RavenAnimationDataCurveDouble)
#endif
